manager/host/skyline.go

```diff
@@ -0,0 +1,239 @@
+	}
+}
+
+func (req *QueryRequest) SetNum(num int) {
+	req.Page = 1
+	req.Num = num
+}
+
+func (req *QueryRequest) SetSelect(sel string) {
+	req.Select = sel
+}
+
+func (req *QueryRequest) SetFrom(from string) {
+	req.From = from
+}
+
+func (req *QueryRequest) SetCondition(cond string) {
+	req.Condition = cond
+}
+
+type Item struct {
+	Ip                 string   `json:"ip"`
+	Sn                 string   `json:"sn"`
+	HostName           string   `json:"host_name"`
+	VpcId              string   `json:"vpc_id"`
+	SecurityDomain     string   `json:"security_domain"`
+	IdcAbbreviation    string   `json:"idc.abbreviation"`
+	IdcCountry         string   `json:"idc.country"`
+	IdcArea            string   `json:"idc.area"`
+	IdcProvince        string   `json:"idc.province"`
+	IdcCity            string   `json:"idc.city"`
+	RoomAbbreviation   string   `json:"room.abbreviation"`
+	CabinetCabinetNum  string   `json:"cabinet.cabinet_num"`
+	CabinetLogicRegion string   `json:"cabinet.logic_region"`
+	DswClusterName     string   `json:"dsw_cluster.name"`
+	NetAswName         string   `json:"net_asw.name"`
+	LogicPodName       string   `json:"logic_pod.name"`
+	NetPodName         string   `json:"net_pod.name"`
+	Tags               []string `json:"tags"`
+}
+
+type Value struct {
+	TotalCount int     `json:"totalCount"`
+	HasMore    bool    `json:"hasMore"`
+	ItemList   []*Item `json:"itemList"`
+}
+
+type QueryResponse struct {
+	Success      bool   `json:"success"`
+	Value        *Value `json:"value"`
+	ErrorCode    int    `json:"errorCode"`
+	ErrorCodeMsg string `json:"errorCodeMsg"`
+	ErrorMessage string `json:"errorMessage"`
+	AllowRetry   bool   `json:"allowRetry"`
+}
+
+type Auth struct {
+	AppName   string `json:"appName"`
+	Account   string `json:"account"`
+	Timestamp int64  `json:"timestamp"`
+	Signature string `json:"signature"`
+}
+
+type skylineClient struct {
+	domain    string
+	appName   string
+	account   string
+	accessKey string
+}
+
+type ItemSearchRequest struct {
+	Auth *Auth         `json:"auth"`
+	Item *QueryRequest `json:"item"`
+}
+
+func (client *skylineClient) getSignature(timestamp string) string {
+	str := client.account + client.accessKey + timestamp
+	h := md5.New()
+	h.Write([]byte(str))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func (client *skylineClient) itemSearch(req *QueryRequest) (*QueryResponse, error) {
+	now := time.Now().Unix()
+	body, err := json.Marshal(&ItemSearchRequest{
+		Auth: &Auth{
+			AppName:   client.appName,
+			Account:   client.account,
+			Timestamp: now,
+			Signature: client.getSignature(strconv.FormatInt(now, 10)),
+		},
+		Item: req,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("%s/item/query", client.domain)
+	request, err := http.NewRequest("POST", url, bytes.NewReader(body))
+	defer request.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Set("Content-Type", "application/json;charset=UTF-8")
+	request.Header.Set("account", client.account)
```

Comment:
What is `account` header?
If you use a custom header, prefix it with `x-`.

Reply:
specd by skyline(cmdb) header, not manager custom header